src/node/extension/extension.ts

```diff
@@ -388,15 +386,31 @@ function startSession(config: any): StartSessionResult {
 		}
 	}
 
+	let determineDebugTypeP: Promise<string|null>;
+	if (config.request === 'attach' && typeof config.processId === 'string' && config.processId.trim() === '${command:PickProcess}') {
+		determineDebugTypeP = pickProcessAndDetermineDebugType(config).then(debugType => {
+			if (debugType) {
+				config.processId = undefined;
+				config.port = debugType === 'node2' ? INSPECTOR_PORT_DEFAULT : LEGACY_PORT_DEFAULT;
+			}
+
+			return debugType;
```

Comment:
Error reporting needs work, but I don't know how to handle that particular case, because the `process.kill` call doesn't fail. I think the only sign of it failing is the stderr message which we can't read in this case.